be/src/olap/column_file/segment_reader.cpp

```diff
@@ -775,6 +775,14 @@ OLAPStatus SegmentReader::_read_all_data_streams(size_t* buffer_size) {
             continue;
         }
 
+        //skip unnecessary stream init
+        //if the query doesn't inclde this column, we needn't init the stream for this column
+        //to reduce the call times for ByteBuffer::create and decrease the TCMalloc pressure
+        if (!_is_column_included(unique_column_id)) {
+            *buffer_size += OLAP_DEFAULT_COLUMN_STREAM_BUFFER_SIZE + sizeof(StreamHead);
+            continue;
+        }
+
```

Comment:
it is done in line 768~771 and there is no need to add buffer_size if the column is not included